server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -361,41 +384,65 @@ private void loadAllLookupsAndInitStateRef()
   private List<LookupBean> getLookupListFromCoordinator(String tier)
   {
     try {
-      final FullResponseHolder response = fetchLookupsForTier(tier);
-      List<LookupBean> lookupBeanList = new ArrayList<>();
-      if (!response.getStatus().equals(HttpResponseStatus.OK)) {
-        LOG.error(
-            "Error while fetching lookup code from Coordinator with status[%s] and content[%s]",
-            response.getStatus(),
-            response.getContent()
-        );
-        return null;
-      }
-
-      // Older version of getSpecificTier returns a list of lookup names.
-      // Lookup loading is performed via snapshot if older version is present.
-      // This check is only for backward compatibility and should be removed in a future release
-      if (response.getContent().startsWith("[")) {
-        LOG.info("Failed to retrieve lookup information from coordinator. Attempting to load lookups using snapshot instead");
-        return null;
-      } else {
-        Map<String, LookupExtractorFactoryContainer> lookupMap = jsonMapper.readValue(
-            response.getContent(),
-            LOOKUPS_ALL_REFERENCE
-        );
+      boolean[] firstAttempt = new boolean[] {true};
+      Map<String, LookupExtractorFactoryContainer> lookupMap = RetryUtils.retry(
+          () -> {
+            if (firstAttempt[0]) {
+              firstAttempt[0] = false;
+            } else {
+              // Wait 1 minute between coordinator fetches
```

Comment:
This should be "add an extra minute in addition to the retry wait"

Since retry utils has a max retry wait of 1 minute, this is essentially guaranteeing at least a 1 minute wait.

Reply:
Changed comment